scipy/spatial/transform/_rigidtransformation.pyx

```diff
@@ -0,0 +1,1899 @@
+    (especially in computer graphics), but the core concept is the same: a
+    rotation and translation describing the orientation of one 3D coordinate
+    frame relative to another. Mathematically, these transformations belong to
+    the Special Euclidean group SE(3), which encodes rotation (SO(3)) plus
+    translation.
+
+    The following operations on rigid transformations are supported:
+
+    - Application on vectors
+    - Transformation composition
+    - Transformation inversion
+    - Transformation indexing
+
+    Note that reflections are not supported, and coordinate systems must be
+    right-handed. Because of this, this class more precisely represents
+    *proper* rigid transformations in SE(3) rather than rigid transformations
+    in E(3) more generally [1]_.
+
+    Indexing within a transformation is supported since multiple
+    transformations can be stored within a single `RigidTransformation`
+    instance.
+
+    To create `RigidTransformation` objects use ``from_...`` methods
+    (see examples below). ``RigidTransformation(...)`` is not supposed
+    to be instantiated directly.
+
+    For rigorous introductions to rigid transformations, see [2]_, [3]_, and
+    [4]_.
+
+    Attributes
+    ----------
+    single
+    rotation
+    translation
+
+    Methods
+    -------
+    __len__
+    __getitem__
+    __mul__
+    __pow__
+    from_rotation
+    from_translation
+    from_matrix
+    from_rottrans
+    from_expcoords
+    from_dualquat
+    as_matrix
+    as_rottrans
+    as_expcoords
+    as_dualquat
+    concatenate
+    apply
+    inv
+    identity
+    random
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Rigid_transformation
+    .. [2] https://motion.cs.illinois.edu/RoboticSystems/CoordinateTransformations.html
+    .. [3] https://www.brainvoyager.com/bv/doc/UsersGuide/CoordsAndTransforms/SpatialTransformationMatrices.html
+    .. [4] Kevin M. Lynch and Frank C. Park, "Modern Robotics: Mechanics,
+           Planning, and Control" Chapter 3.3, 2017, Cambridge University Press.
+           https://hades.mech.northwestern.edu/images/2/25/MR-v2.pdf#page=107.31
+    .. [5] Paul Furgale, "Representing Robot Pose: The good, the bad, and the
+           ugly", June 9, 2014.
+           https://rpg.ifi.uzh.ch/docs/teaching/2024/FurgaleTutorial.pdf
+
+    Notes
+    -----
+    .. versionadded:: 1.16.0
+
+    Examples
+    --------
+
+    A `RigidTransformation` instance can be initialized in any of the
+    above formats and converted to any of the others. The underlying object is
+    independent of the representation used for initialization.
+
+    **Notation Conventions and Composition**
+
+    The notation here largely follows the convention defined in [5]_.
+    When we name transformations, we read the subscripts from right to left.
+    So ``t_A_B`` represents a transformation A <- B and can be interpreted as:
+
+    - the coordinates of B relative to A
```

Comment:
Does it applicable when `t` is not a mere translation? As I understand this paragraph you are talking about a general rigid transformation here.

Reply:
Updated to "coordinates and orientation"